### PR TITLE
Copy edit the Ltac2 documentation

### PR DESCRIPTION
**Pull request by tchajed**
I read through the Ltac2 documentation and did some copy editing along the way. I intentionally avoided making any changes I thought would be controversial, so I think all of these changes either fix grammar or sound more natural.

**Kind:** documentation

### CODE REVIEW

File: doc/sphinx/proof-engine/ltac2.rst [hunk 1]
```diff
@@ -923,8 +923,8 @@ Due to conflicts, a few syntactic rules have changed.
 
 - The dispatch tactical :n:`tac; [foo|bar]` is now written :n:`tac > [foo|bar]`.
 - Levels of a few operators have been revised. Some tacticals now parse as if
-  they were a normal function, i.e. one has to put parentheses around the
-  argument when it is complex, e.g an abstraction. List of affected tacticals:
+  they were a normal function; that is, one has to put parentheses around the
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst]: were normal functions.  Parentheses are now required around complex arguments such as abstractions.

**Review by jfehrle — APPROVED** [review]

**Comment by jfehrle** [comment]
Looks good.  I suggested a few additional changes.

File: doc/sphinx/proof-engine/ltac2.rst [hunk 2]
```diff
@@ -315,36 +315,36 @@ Fatal errors
...
-through the following primitive in module `Control`.::
+through the following primitive in module `Control`::
 
   val throw : exn -> 'a
 
 Unlike backtracking exceptions from the next section, this kind of error
 is never caught by backtracking primitives, that is, throwing an exception
-destroys the stack. This is materialized by the following equation, where `E`
-is an evaluation context.::
+destroys the stack. This is exemplified by the following equation, where `E`
```
[the remark this answers is not in the document]
Comment (tchajed) [reply doc/sphinx/proof-engine/ltac2.rst]: I changed this to "codified", which is closer to the word I was looking for. It's not just an example but formally defines what the text says.

File: doc/sphinx/proof-engine/ltac2.rst [hunk 3]
```diff
@@ -315,36 +315,36 @@ Fatal errors
...
+is an evaluation context::
 
   E[throw e] ≡ throw e
 
   (e value)
 
-There is currently no way to catch such an exception and it is a design choice.
-There might be at some future point a way to catch it in a brutal way,
+There is currently no way to catch such an exception, a deliberate design choice.
+Eventually there might be a way to catch it in a brutal way,
```
[the remark this answers is not in the document]
Comment (tchajed) [reply doc/sphinx/proof-engine/ltac2.rst]: good call, removed

File: doc/sphinx/proof-engine/ltac2.rst (hunk 1 above)
Comment (tchajed) [reply doc/sphinx/proof-engine/ltac2.rst]: yes, that is much better

**Review by ppedrot — APPROVED** [review]

**Pull request merged**